schema_cache.rst

```diff
@@ -0,0 +1,111 @@
+.. _schema_cache:
+
+Schema Cache
+============
+
+There are several PostgREST features that need information from the database schema.
+If they accessed this information directly from the database every time they needed it, it would be too costly.
+That is why, when PostgREST starts, it generates a database schema cache and uses it to get the information needed for these features:
+
+- For :ref:`resource_embedding`, it looks for foreign key constraints to determine relationships between tables.
+- For :ref:`s_procs`, it looks for their metadata: parameters, return type, volatility and `overloading <https://www.postgresql.org/docs/current/xfunc-overload.html>`_.
+- For :ref:`upsert` statements, it looks for the tables' primary keys.
+- For INSERT statements, it also looks for the tables' primary keys in order to return the Location header.
```

Comment:
This one should link to https://postgrest--404.org.readthedocs.build/en/404/api.html#insertions-updates